samples/net/sockets/echo_client/src/echo-client.c

```diff
@@ -327,10 +333,11 @@ int main(void)
 	k_thread_access_grant(k_current_get(), &run_app);
 	k_mem_domain_add_thread(&app_domain, k_current_get());
 
-	k_thread_user_mode_enter((k_thread_entry_t)start_client, NULL, NULL,
+	k_thread_user_mode_enter(start_client, &result, NULL,
```

Comment:
Similar comment here as with mqtt_sn_publisher. Since _result_ might not be accessible from user space, we should probably use _NULL_ for this case and update _start_client()_ as appropriate.

Reply:
I'm handling this similarly to the mqtt_sn_publisher sample